crates/viewer/re_viewer_context/src/cache/image_decode_cache.rs

```diff
@@ -115,6 +167,135 @@ fn decode_image(
+            decoded_format.color_model
+        )));
+    }
+
+    if decoded_format.datatype() != format.datatype() {
+        return Err(ImageLoadError::DecodeError(format!(
+            "Encoded depth PNG datatype mismatch: blob is {:?}, expected {:?}",
+            decoded_format.datatype(),
+            format.datatype()
+        )));
+    }
+
+    let expected_num_bytes = format.num_bytes();
+    let ImageBuffer(blob) = buffer;
+    let actual_num_bytes = blob.len();
+    if actual_num_bytes != expected_num_bytes {
+        return Err(ImageLoadError::DecodeError(format!(
+            "Encoded depth PNG payload is {actual_num_bytes} B, but {} requires {expected_num_bytes} B",
+            format.0
+        )));
+    }
+
+    Ok(ImageInfo::from_stored_blob(
+        blob_row_id,
+        blob_component,
+        blob,
+        format.0,
+        ImageKind::Depth,
+    ))
+}
+
+fn decode_rvl_depth(
+    blob_row_id: RowId,
+    blob_component: ComponentIdentifier,
+    image_bytes: &[u8],
+    format: &ImageFormatComponent,
+) -> Result<ImageInfo, ImageLoadError> {
+    let metadata = parse_ros_rvl_metadata(image_bytes)
+        .map_err(|err| ImageLoadError::DecodeError(err.to_string()))?;
+
+    let expected_pixels = (format.width as usize) * (format.height as usize);
+    if metadata.num_pixels() != expected_pixels {
+        return Err(ImageLoadError::DecodeError(format!(
+            "RVL encoded depth metadata {metadata_width}x{metadata_height} disagrees with ImageFormat {}x{}",
+            format.width,
+            format.height,
+            metadata_width = metadata.width,
+            metadata_height = metadata.height
+        )));
+    }
+
+    let buffer: Vec<u8> = match format.datatype() {
+        ChannelDatatype::U16 => decode_ros_rvl_u16(image_bytes, &metadata)
+            .map(|x: Vec<u16>| vec_into_bytes(&x))
+            .map_err(|err| ImageLoadError::DecodeError(err.to_string()))?,
+        ChannelDatatype::F32 => decode_ros_rvl_f32(image_bytes, &metadata)
+            .map(|x: Vec<f32>| vec_into_bytes(&x))
+            .map_err(|err| ImageLoadError::DecodeError(err.to_string()))?,
+        other => {
+            return Err(ImageLoadError::DecodeError(format!(
+                "Unsupported RVL channel datatype {other:?}"
+            )));
+        }
+    };
+
+    let expected_num_bytes = format.num_bytes();
+    let actual_num_bytes = buffer.len();
+    if actual_num_bytes != expected_num_bytes {
+        return Err(ImageLoadError::DecodeError(format!(
+            "RVL payload decoded to {actual_num_bytes} B, but {} requires {expected_num_bytes} B",
+            format.0
+        )));
+    }
+
+    Ok(ImageInfo::from_stored_blob(
+        blob_row_id,
+        blob_component,
+        Blob::from(buffer),
+        format.0,
+        ImageKind::Depth,
+    ))
+}
+
+fn vec_into_bytes<T: Pod>(vec: &[T]) -> Vec<u8> {
+    bytemuck::cast_slice(vec).to_vec()
+}
```

Comment:
the callsites should do direct vector conversion instead of doing a full memcpy to a new vec, use https://docs.rs/bytemuck/latest/bytemuck/allocation/fn.cast_vec.html